Source/WebCore/html/HTMLInputElement.idl

```diff
@@ -82,7 +83,7 @@
     [ImplementedAs=selectionDirectionForBindings] attribute DOMString? selectionDirection;
 
     undefined setRangeText(DOMString replacement);
-    undefined setRangeText(DOMString replacement, unsigned long start, unsigned long end, optional DOMString selectionMode);
+    undefined setRangeText(DOMString replacement, unsigned long start, unsigned long end, optional DOMString selectionMode = "preserve");
```

Comment:
You need to change DOMString to the SelectionMode enum as well (and define that enum). And then also update the corresponding h and cpp files for that enum. Look at `BinaryType` in WebSocket for an idea of what that might entail.

Hope that helps!

(Also very surprising there is no WPT coverage. Might be worth adding as well if you're up for it.)

Reply:
Yes - I am doing more changes locally. I tried to do simple 'SelectionMode' enum change and it didn't work and failed to compile. So I will look into 'WebSocket' PR on how it was added for more help / reference. Thanks!